src/pages/terraform/[...allDocs].tsx

```diff
@@ -0,0 +1,237 @@
+import terraformData from 'data/terraform.json'
+import { ProductData } from 'types/products'
+import { getStaticGenerationFunctions } from 'layouts/sidebar-sidecar/server'
+import DocsView from 'views/docs-view'
+
+// function DocsView(props) {
+// 	return (
+// 		<>
+// 			<pre>
+// 				<code>{JSON.stringify(props, null, 2)}</code>
+// 			</pre>
+// 		</>
+// 	)
+// }
+
+const PRODUCT_DATA = terraformData as ProductData
+
+const CUSTOM_DOCS_LANDING_PATHS = ['docs']
+
+/*
+Subpath				      Repository
+----------------------------------------------
+/cdktf			      	terraform-cdk
+/cli		  	      	terraform
+/cloud-docs		    	terraform-docs-common
+/cloud-docs/agents	terraform-docs-agents
+/configuration	  	terraform
+/docs			        	terraform
+/enterprise		    	ptfe-releases
+/guides		      		terraform
+/internals	    		terraform
+/intro		      		terraform
+/language		      	terraform
+/plugin		      		terraform-docs-common
+/plugin/framework   terraform-plugin-framework
+/plugin/log		    	terraform-plugin-log
+/plugin/mux	    		terraform-plugin-mux
+/plugin/sdkv2   		terraform-plugin-sdk
+/registry			      terraform-docs-common
+*/
+const TERRAFORM_DOCS_SUBPATHS = {
+	cdktf: {
+		baseName: 'CDKTF',
+		productSlugForLoader: 'terraform-cdk',
+	},
+	cli: {
+		baseName: 'CLI',
+		productSlugForLoader: 'terraform',
+	},
+	'cloud-docs': {
+		baseName: 'Cloud Docs',
+		productSlugForLoader: 'terraform-docs-common',
+	},
+	'cloud-docs/agents': {
+		baseName: 'Cloud Docs Agents',
+		productSlugForLoader: 'terraform-docs-agents',
+		navDataPrefix: 'cloud-docs-agents',
+	},
+	configuration: {
+		baseName: 'Configuration',
+		productSlugForLoader: 'terraform',
+	},
+	docs: {
+		baseName: 'Docs',
+		productSlugForLoader: 'terraform',
+	},
+	enterprise: {
+		baseName: 'Enterprise',
+		productSlugForLoader: 'ptfe-releases',
+	},
+	guides: {
+		baseName: 'Guides',
+		productSlugForLoader: 'terraform',
+	},
+	internals: {
+		baseName: 'Internals',
+		productSlugForLoader: 'terraform',
+	},
+	intro: {
+		baseName: 'Intro',
+		productSlugForLoader: 'terraform',
+	},
+	language: {
+		baseName: 'Language',
+		productSlugForLoader: 'terraform',
+	},
+	plugin: {
+		baseName: 'Plugin',
+		productSlugForLoader: 'terraform-docs-common',
+	},
+	'plugin/framework': {
+		baseName: 'Plugin Framework',
+		productSlugForLoader: 'terraform-plugin-framework',
+		navDataPrefix: 'plugin-framework',
+	},
+	'plugin/log': {
+		baseName: 'Plugin Log',
+		productSlugForLoader: 'terraform-plugin-log',
+		navDataPrefix: 'plugin-log',
+	},
+	'plugin/mux': {
+		baseName: 'Plugin mux',
+		productSlugForLoader: 'terraform-plugin-mux',
+		navDataPrefix: 'plugin-mux',
+	},
+	'plugin/sdkv2': {
+		baseName: 'Plugin sdkv2',
+		productSlugForLoader: 'terraform-plugin-sdk',
+		navDataPrefix: 'plugin-sdk',
+	},
+	registry: {
+		baseName: 'Registry',
+		productSlugForLoader: 'terraform-docs-common',
+	},
+}
+
+async function getSubpathStaticPaths({
```

Comment:
I'd love to see a test case for this function so that I (or anyone) can easily grok the inputs/outputs without needing to jump into the implementation here immediately, as well as be able quickly know if any future refactoring accidentally changes the behavior.

Reply:
Have split out the utilities that this stuff evolved into, and added some tests - let me know what you think 🙇‍♂️